src/mono/CMakeLists.txt

```diff
@@ -604,6 +611,14 @@ if(GCC)
   endif()
 endif()
 
+TEST_BIG_ENDIAN(IS_BIG_ENDIAN)
+
+if(IS_BIG_ENDIAN)
+  set(TARGET_BYTE_ORDER G_BIG_ENDIAN)
+else()
+  set(TARGET_BYTE_ORDER G_LITTLE_ENDIAN)
+endif()
+
 if(IS_BIG_ENDIAN)
   set(ORDER G_BIG_ENDIAN)
```

Comment:
as far as I can see we only use `ORDER` in src/mono/cmake/eglib-config.h.cmake.in, while the rest of mono uses `TARGET_BYTE_ORDER`. please clean this up to use the latter in the eglib config file so we can delete this block.

looks like src/mono/mono/eglib/eglib-config.h.in is unused too since it was replaced by the /eglib-config.h.cmake.in file dotnet/runtime.

Reply:
Hmm actually:
https://github.com/dotnet/runtime/blob/d5668e7e084a0bf51c146cbdb0ec8bb0a42a6155/src/mono/CMakeLists.txt#L748

Sounds like this needs a bit more cleanup, we can do it in another PR 😄

Reply:
This comment seems to be incorrect, at least for Linux, see mono/mono/eglib/glib.h:
```
#ifdef _MSC_VER
#include <eglib-config.hw>
#else
#include <eglib-config.h>
#endif

```

With this fix, I do get the correct byte order define in both places:
```
artifacts/obj/mono/Linux.s390x.Debug/mono/eglib/eglib-config.h:#define G_BYTE_ORDER             G_BIG_ENDIAN
artifacts/obj/mono/Linux.s390x.Debug/config.h:#define TARGET_BYTE_ORDER G_BIG_ENDIAN

```

Just to confirm: Did you now want me to change anything, or is the patch OK (at least for now)?

Reply:
ok, thanks for looking, makes sense. I think we can cleanup the eglib-config situation now since there should be no difference between the Windows config and other platforms now that we're using CMake instead of autotools.

that's unrelated to your PR though so I'll handle it separately :)